tss-esapi/src/interface_types/structure_tags.rs

```diff
@@ -78,54 +79,44 @@ impl TryFrom<TPMI_ST_ATTEST> for AttestationType {
 impl Marshall for AttestationType {
     const BUFFER_SIZE: usize = std::mem::size_of::<TPMI_ST_ATTEST>();
 
-    fn marshall_offset(
-        &self,
-        marshalled_data: &mut [u8],
-        offset: &mut std::os::raw::c_ulong,
-    ) -> Result<()> {
+    fn marshall_offset(&self, marshalled_data: &mut [u8], offset: &mut usize) -> Result<()> {
+        let ffi_buffer_size = FfiSizeType::try_from(marshalled_data.len())?;
```

Comment:
This is going to be such a common pattern - what do you think about implementing a macro in `ffi.rs` to go from `&mut [u8]` and `&mut usize` to `size_t`, via `FfiSizeType`? And you can just use that within the MU call.

Reply:
I think the pattern is so common it would probably be possible the entire implementation of marshall and unmarshall in a macro.

Reply:
I think I will look into if it is possible to just do a proc_macro
```rust
#[derive(marshall(TPM2B_DIGEST), unmarshall(TPM2B_DIGEST)]
```